src/NativeBase/Components/TagButtons.tsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react'
+import { HStack, Icon, IconButton, Pressable, Text } from 'native-base'
+import MaterialIcons from 'react-native-vector-icons/MaterialIcons'
+
+interface TagButtonsProps {
+  tags: string[] // An array of tag names, e.g., ['Roles', 'Tech', 'Causes']
+  iconState: Record<string, boolean> // A dynamic record that maps tags to boolean states
+  handleTagPress: (tag: string) => void // A function that takes any tag
+}
+
+const TagButtons: React.FC<TagButtonsProps> = ({
+  tags,
+  iconState,
+  handleTagPress,
+}) => {
+  return (
+    <HStack space={2} alignItems="center">
+      {tags.map(tag => (
+        <Pressable
```

Comment:
I'd also extract a single tagbutton into it's own component. This component here (TagButtons) could be then just using those tagbutton components. Naming wise we could also name HTagButtons (H for horinzontal) but as we do have only one version currently I think FilterTagButtons would be a good name as it says that it's about tag buttons used for filtering.